### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by jimflores5**

### CODE REVIEW

File: src/assignments/tech-jobs-mvc.rst [hunk 1]
```diff
@@ -3,4 +3,510 @@
...
+
+Your first two tasks as an apprentice went well! You, Blake, and Sally built
+the TechJobs console prototype and then refactored the code to move it to an
+object-oriented format.
+
+After demonstrating the prototype for the Company Team at LaunchCode, it
+received the green light to be fully built out as a web application.
+
+The first step in this process will be to quickly develop a `minimum viable
+product <https://en.wikipedia.org/wiki/Minimum_viable_product>`__, or MVP. The
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst:19]: Do you think "minimum viable product" should be added to the index?

File: src/assignments/tech-jobs-mvc.rst [hunk 2]
```diff
@@ -3,4 +3,510 @@
...
+   work).
+#. From the welcome screen, click *Check out from Version Control*, select
+   *Git*.
+#. Choose your fork from the repository dropdown, select the parent directory
+   where you want to store your project, and hit *Clone*.
+#. In the screens that follow:
+
+   a. Choose *Create Project From Existing Sources* on the first pane.
+   b. Select *Auto Import* in the Gradle configuration pane.
+   c. Use the defaults on all other panes.
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst]: Are these instructions impacted by the discussion during the working meeting earlier today?

File: src/assignments/tech-jobs-mvc.rst [hunk 3]
```diff
@@ -3,4 +3,510 @@
...
+
+#. Start up the application (via the Gradle pane, *Tasks > Application >
+   bootRun*), so you can refer to both the code as well as the running app
+   while we look at what’s in place already.
+
+.. admonition:: Note
+
+   You’ll spend a lot of time reading and understanding the code that’s provided
+   for you. This is OK! It’s an essential skill for a programmer to be
+   able to read and understand code written by other developers.
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst]: I would rephrase this note. Maybe, "It's an essential skill for a programmer to be able to read and understand code written by other developers. Make sure you have read and understood the code that's provided for you!"

File: src/assignments/tech-jobs-mvc.rst [hunk 4]
```diff
@@ -3,4 +3,510 @@
...
+
+The ``HomeController``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+This class has only one handler method, ``index``, which displays the home page
+for the app. The controller renders the ``index.html`` template (in
+``src/main/resources/templates``).
+
+If you haven’t done so already, launch the app and examine the home page to see
+what it looks like.
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst]: I would give them a reminder to immediately launch the app with the starter code and see what it looks like. Will the starter code even run? What should they see? It might be helpful for people to understand the starter code.

File: src/assignments/tech-jobs-mvc.rst [hunk 5]
```diff
@@ -3,4 +3,510 @@
...
+.. admonition:: Tip
+
+   We use `Twitter’s Bootstrap <http://getbootstrap.com/>`__ CSS, HTML, and
+   JS framework to provide some styling and functionality to our views. The
+   appropriate files are included at the top of ``fragments.html`` and thus
+   are included on every page of our app.
+
+   You won’t have to explicitly use Bootstrap at all in this assignment,
+   but it’s a great way to make your sites look good with minimal work.
+   Consider using it on your own projects!
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst]: "Consider using it in your own projects!"

**Review by gildedgardenia — COMMENTED** [review]
I think it is looking pretty good! I had some minor phrasing questions, but that was it!

File: src/assignments/tech-jobs-mvc.rst (hunk 1 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst:19]: Done.  Thanks.

File: src/assignments/tech-jobs-mvc.rst (hunk 2 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst]: Yes they are.  I've updated these steps to reflect the new sequence.  Could you please verify that they accurately describe what the students need to do?

File: src/assignments/tech-jobs-mvc.rst (hunk 3 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst]: I reworded this admonition, changed it to a "Tip", and shifted it down into the following section.  Let me know what you think.

File: src/assignments/tech-jobs-mvc.rst (hunk 4 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst]: The starter code does run on its own, and I reworded the last Getting Started step to emphasize launching the app.  I also added a screenshot of the home page.

File: src/assignments/tech-jobs-mvc.rst (hunk 5 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst]: Fixed.  Thanks!

File: src/assignments/tech-jobs-mvc.rst [hunk 6]
```diff
@@ -3,4 +3,509 @@
...
+
+For this next step in the project, you’ll be working with Carly.
+
+.. figure:: figures/LC-Carly.png
+   :scale: 55%
+   :alt: Carly's LaunchCode avatar.
+
+Carly was once a LaunchCode apprentice as well, so she knows just what
+it’s like to be in your shoes. She’s done some initial work on the
+project and left you some TODOs that she knows you can handle.
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst]: Should "TODOs" be code?

File: src/assignments/tech-jobs-mvc.rst [hunk 7]
```diff
@@ -3,4 +3,509 @@
...
+   and understand someone else's code. This assignment begins with you
+   practicing exactly that. Make sure you carefully examine the provided code
+   BEFORE you start changing things.
+
+   Trying to "fix" a code sample before understanding how it works leads to
+   confusion, frustration, and a broken program. DO NOT SKIP the code review!
+
+Carly created a Spring Boot application and filled in some features. She
+refactored ``JobData`` to generate an ArrayList of ``Job`` objects based on
+your TechJobs-OO work, and she added controllers and views for a Home, List and
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst]: Oxford comma

Comment (gildedgardenia) [reply src/assignments/tech-jobs-mvc.rst]: Also, it might be helpful to highlight these page names in quotation marks or as code.

File: src/assignments/tech-jobs-mvc.rst [hunk 8]
```diff
@@ -3,4 +3,509 @@
...
+Search page. ``JobData`` now also builds ArrayLists for the ``Employer``,
+``Location``, ``PositionType``, and ``CoreCompetency`` objects.
+
+The Model
+^^^^^^^^^^
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst]: One extra carrot!

File: src/assignments/tech-jobs-mvc.rst [hunk 9]
```diff
@@ -3,4 +3,509 @@
...
+#. Be sure to test your code by running the program and clicking links from
+   different categories on the ``/list`` page.
+
+   a. Clicking the *Web - Back End* link yields 6 jobs.
+   b. *iOS* yields 4 jobs.
+   c. *New York* yields 1 job.
+   d. Etc.
+
+Add ``View All`` Link
+^^^^^^^^^^^^^^^^^^^^^^
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst]: Extra carrot!

File: src/assignments/tech-jobs-mvc.rst [hunk 10]
```diff
@@ -3,4 +3,509 @@
...
+   pair. Check the method that renders the template to help identify the name
+   to use for the key.
+#. Modify ``list.html`` to fill in the empty table cell with the necessary
+   link. Check ``ListController`` to help identify the data to pass in for the
+   query parameters.
+
+   .. admonition:: Warning
+
+      This is an either/or situation. You only need to implement ONE option, not
+      both.
```
Comment (gildedgardenia) [inline src/assignments/tech-jobs-mvc.rst]: Do you mean that they need to pick between steps 1 or 2? You should add that specifically to avoid confusion.

**Review by gildedgardenia — APPROVED** [review]
A few small typos, but it looks great! Approved!

File: src/assignments/tech-jobs-mvc.rst (hunk 6 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst]: Fixed.  Thanks.

File: src/assignments/tech-jobs-mvc.rst (hunk 7 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst]: Quotes added, comma fixed.

File: src/assignments/tech-jobs-mvc.rst (hunk 8 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst]: Removed.  Thanks.  (Does it really matter?)

File: src/assignments/tech-jobs-mvc.rst (hunk 9 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst]: Extraneous veggie consumed.

File: src/assignments/tech-jobs-mvc.rst (hunk 10 above)
Comment (jimflores5) [reply src/assignments/tech-jobs-mvc.rst]: I fixed the phrasing to make this point more obvious.
